src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -55,133 +54,132 @@
+                errorMap.put(subscriptionName, e.getMessage());
+            }
+        });
+        return (errorMap.isEmpty()) ? new ResponseEntity<>(HttpStatus.OK) : new ResponseEntity<>(getSubscriptionResponseList(errorMap), HttpStatus.BAD_REQUEST);
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Removes the subscription from the database")
-    public ResponseEntity<SubscriptionResponse> deleteSubscriptionById(@PathVariable String subscriptionName) {
-        SubscriptionResponse subscriptionResponse = new SubscriptionResponse();
-        LOG.info("Subscription :" + subscriptionName + " delete started");
-        if (subscriptionService.deleteSubscription(subscriptionName)) {
-            LOG.info("Subscription :" + subscriptionName + " deleted Successfully");
-            subscriptionResponse.setMsg("Deleted Successfully");
-            subscriptionResponse.setStatusCode(HttpStatus.OK.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.OK);
-        } else {
-            LOG.info("Subscription :" + subscriptionName + " delete completed :: Record not found for delete");
-            subscriptionResponse.setMsg("Record not found for delete");
-            subscriptionResponse.setStatusCode(HttpStatus.BAD_REQUEST.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.BAD_REQUEST);
-        }
+    @ApiOperation(value = "Removes the subscriptions from the database")
+    public ResponseEntity<List<SubscriptionResponse>> deleteSubscriptionById(@PathVariable String subscriptionName) {
+        errorMap = new HashMap<>();
+        List<String> subscriptionNames = Arrays.asList(subscriptionName.split(","));
+
+        subscriptionNames.forEach(name -> {
+            LOG.debug("Subscription delete started :: " + name);
+            if (subscriptionService.deleteSubscription(name)) {
+                LOG.debug("Subscription was deleted successfully :: " + name);
+            } else {
+                LOG.error("Subscription was not found :: " + name);
+                errorMap.put(name, "Subscription was not found");
+            }
+        });
+        return (errorMap.isEmpty()) ? new ResponseEntity<>(HttpStatus.OK) : new ResponseEntity<>(getSubscriptionResponseList(errorMap), HttpStatus.BAD_REQUEST);
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Retrieve all the subscriptions")
+    @ApiOperation(value = "Retrieves all the subscriptions")
     public ResponseEntity<List<Subscription>> getSubscriptions() {
-        LOG.info("Subscription : get all records started");
-        List<Subscription> subscriptionList = new ArrayList<>();
+        LOG.debug("Subscription get all records started");
         try {
-            subscriptionList = subscriptionService.getSubscription();
-            return new ResponseEntity<>(subscriptionList, HttpStatus.OK);
+            return new ResponseEntity<>(subscriptionService.getSubscription(), HttpStatus.OK);
```

Comment:
getSubscription() should be refactored and renamed to getSubscriptions() since it retrieves one or several subscriptions.